ClientTxnMultiMapTest.testConcrruentTxnPut

```
java.lang.AssertionError: expected null, but was:<com.hazelcast.transaction.TransactionNotActiveException: Transaction is not active>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at org.junit.Assert.assertNull(Assert.java:747)
	at com.hazelcast.client.txn.ClientTxnMultiMapTest.testConcrruentTxnPut(ClientTxnMultiMapTest.java:144)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-ZuluJDK6-freestyle/10/testReport/junit/com.hazelcast.client.txn/ClientTxnMultiMapTest/testConcrruentTxnPut/

```
java.lang.AssertionError: expected null, but was:<com.hazelcast.transaction.TransactionNotActiveException: Transaction is not active!>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at org.junit.Assert.assertNull(Assert.java:747)
	at com.hazelcast.client.txn.ClientTxnMultiMapTest.testConcrruentTxnPut(ClientTxnMultiMapTest.java:144)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-freestyle/127/testReport/junit/com.hazelcast.client.txn/ClientTxnMultiMapTest/testConcrruentTxnPut/

```
java.lang.AssertionError: expected null, but was:<com.hazelcast.transaction.TransactionException: No transaction context found for txnId:ad2807ab-c3ae-4ada-b2ad-2a470d144c1f>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at org.junit.Assert.assertNull(Assert.java:747)
	at com.hazelcast.client.txn.ClientTxnMultiMapTest.testConcrruentTxnPut(ClientTxnMultiMapTest.java:144)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-freestyle/32/testReport/junit/com.hazelcast.client.txn/ClientTxnMultiMapTest/testConcrruentTxnPut/

```
java.lang.AssertionError: expected null, but was:<com.hazelcast.transaction.TransactionNotActiveException: No transaction is found while accessing transactional object -> hz:impl:multiMapService[819de9b2-524e-45a3-aea5-6dd46cf0091a]!>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotNull(Assert.java:755)
	at org.junit.Assert.assertNull(Assert.java:737)
	at org.junit.Assert.assertNull(Assert.java:747)
	at com.hazelcast.client.txn.ClientTxnMultiMapTest.testConcrruentTxnPut(ClientTxnMultiMapTest.java:144)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-freestyle/148/testReport/junit/com.hazelcast.client.txn/ClientTxnMultiMapTest/testConcrruentTxnPut/

--- a/hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandomProvider.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandomProvider.java
@@ -24,6 +24,8 @@ import java.util.Random;
  */
 public final class ThreadLocalRandomProvider {
 
+    private static final Random SEED_GENERATOR = new Random();
+
     private static final ThreadLocal<Random> THREAD_LOCAL_RANDOM = new ThreadLocal<Random>();
     private static final ThreadLocal<SecureRandom> THREAD_LOCAL_SECURE_RANDOM = new ThreadLocal<SecureRandom>();
 
@@ -38,7 +40,8 @@ public final class ThreadLocalRandomProvider {
     public static Random get() {
         Random random = THREAD_LOCAL_RANDOM.get();
         if (random == null) {
-            random = new Random();
+            long seed = SEED_GENERATOR.nextLong();
+            random = new Random(seed);
             THREAD_LOCAL_RANDOM.set(random);
         }
         return random;

